neonvm/runner/main.go

```diff
@@ -527,11 +531,46 @@ func main() {
 		qemuCmd = append(qemuCmd, "-incoming", fmt.Sprintf("tcp:0:%d", vmv1.MigrationPort))
 	}
 
+	go terminateQemu(vmSpec.QMP)
 	if err := execFg(QEMU_BIN, qemuCmd...); err != nil {
 		log.Fatal(err)
 	}
 }
 
+func terminateQemu(qmpPort int32) {
+	log.Println("watching OS signals")
+	c := make(chan os.Signal, 1) // we need to reserve to buffer size 1, so the notifier are not blocked
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	<-c
+	log.Println("got signal, sending powerdown command to QEMU")
+
+	for {
+		mon, err := qmp.NewSocketMonitor("tcp", fmt.Sprintf("127.0.0.1:%d", qmpPort), 2*time.Second)
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+
+		if err := mon.Connect(); err != nil {
+			log.Println(err)
+			continue
+		}
+		defer mon.Disconnect()
+
+		qmpcmd := []byte(`{"execute": "system_powerdown"}`)
+		_, err = mon.Run(qmpcmd)
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+
+		log.Println("system_powerdown command sent to QEMU")
+		break
+	}
```

Comment:
We should have some kind of minimum time to wait between loop iterations here so this loop can't _completely_ flood the logs if it's continually failing.